src/modules/auth/hooks/use-auth.ts

```diff
@@ -0,0 +1,23 @@
+import { api } from '#/configs';
+import { useQuery } from '@tanstack/react-query';
+import { ENDPOINTS } from '../constants';
+import { type AxiosAuthResponse } from '../types';
+
+export function useAuth() {
+  async function getUser() {
+    const { data } = await api.get<AxiosAuthResponse>(ENDPOINTS.user);
+
+    if (!Object.keys(data).includes('user')) {
```

Comment:
I think you do not need this checking and throwing. if the response is not successful (not 200-300), the isError property will be true.

Reply:
yeah, I added this because of eslint. checked it once again, didn't complain.. fixed.